app/users/users.go

```diff
@@ -72,11 +78,12 @@ func (s *UserService) createDBUser(id int) (*models.User, error) {
 }
 
 // Retrieve authenticates user with internal-api and retrieves/creates locally stored user.
-func (s *UserService) Retrieve(token string) (*models.User, error) {
+func (s *UserService) Retrieve(q Query) (*models.User, error) {
+	token := q.Token
 	log := s.logger.LogF(monitor.F{"token": token})
```

Comment:
aren't we leaking tokens this way? is this truly necessary?

Reply:
We are not, tokens are filtered out in the logging package in production. It's necessary for debugging / tests.